### DISCUSSION TITLE
关于GRID大作业的若干疑问

### DISCUSSION THREAD

**Discussion by DM0U** [physics-data/faq#155] [Q&A] [answered]
老师，您好。在GRID大作业的编程中我有一些关于最终测评的疑问，希望能够得到解答。
<br/>
最终的测评样例与目前的代码仓库样例会有多少不同？

1. 首先，最终的测评样例是否仍能保证卫星数量为3？<br/>即输入**全部**的`.rsp`文件的名称格式始终保证为`Cube(\d+)_(\d+).rsp`，使得`\1`的值只取1/2/3？
2. 在样例文件的`constant.json`文件中，只有卫星1的赤经值索引为`'CubeSat1_RA '`（**末尾存在空格**），其余参数的末尾均无空格。目前我采用的解决方法是在程序中直接给出各个参数的索引名称，但是我想知道最终的constant.json文件中是否仍然会带有空格？
3. 从物理或者电子工程学上来说，探测器输出的若干个能道（README中为200个）是否能够保证是一致的？
目前本人对数据的存储方式是对每个`Cube(\d+)_(_d+)`，分别记录其对应的能道（即`["EBOUNDS"]`）范围。但我觉得这种记录方法似乎是过度冗余的。一般来说，我们应该更倾向于为同一个卫星甚至同一类型的卫星采用同一套的探测晶体，因此它们的输出能道也应该是一致的。
如果不同探测晶体间能够保证有相同的输出能道，那么是仅限于同一个卫星吗？还是说所有卫星的所有探测晶体都能相同范围的输出能道？
4. 在其它的faq中曾经说过输入数据的等时间间隔只是为了方便。现在想确认最终给到的数据中能否必然保证初始时刻数据是等间距的？
在gif的绘图中，如果原始的数据不能直接满足等时长间距，将需要复杂的数据处理才能处理好gif中各个帧的关系，也无疑会加长`.gif`文件的生成速度。
5. 最终测评数据的中`.rsp`文件的索引名`.columns.info`是否保证相同？
6. 一个探测器的响应应该认为是对其4个探测晶体的结果取平均还是求和，亦或者应该分别将每个探测晶体都做分别的处理？
7. 我们书写的`.py`文件是否需要严格利用`makefile`中给入的`argv`参数？
对于`make energy_response.h5`语句，`Cube*_det*.rsp`模块匹配并输入的参数过于缺乏顺序，如果单纯利用`argv`参数读入`.rsp`文件将十分复杂，是否允许直接在`calculate_response.py`文件中写入固定的文件相对地址？
(具体语句块如下)

```
energy_response.h5: data/matrix/Cube*_det*.rsp light_curve.h5 data/constant.json
	python3 calculate_response.py $^ $@
```

**Comment by Harry-Chen** [comment]
@babychen9816

**Comment by quxnb** [comment, marked as answer]
1. 是的
2. 空格应该是误加的...抱歉
3. 假设所有卫星输出能道相同
4. 是的
5. 是的
6. 没有统一的方法, 在报告中说明你的做法及其合理性即可
7. (emmm, 我觉得不改 Makefile 应该也能做
